File: cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    while (true)
    {
        // Note: pay attention to the overflow
        float dollars = get_float("Change owed: ");
        if (dollars > 0)
        {
            int cents = round(dollars * 100);
            int count = 0;
            int coins[] = {25, 10, 5, 1};
            int arrayLength = sizeof(coins) / sizeof(coins[0]);

            for (int i = 0; i < arrayLength; i++)
            {
                int changes = cents / coins[i];
                cents = cents % coins[i];
                count += changes;
            }

            printf("%i\n", count);
            return 0;
        }
    }
}

File: pset6/cash/cash.py
from cs50 import get_float

while True:
    dollars = get_float("Change owed: ")
    if dollars > 0:
        cents = round(dollars * 100)
        count = 0
        coins = [25, 10, 5, 1]

        for i in coins:
            count += cents // i
            cents = cents % i

        print(count)
        break
